From 1c70b4d2b70155558a9b5edb79a172696c16e129 Mon Sep 17 00:00:00 2001
From: Otavio Salvador <otavio@ossystems.com.br>
Date: Fri, 22 May 2009 18:02:04 -0300
Subject: [PATCH] probe GPIO device before asking for base_addr

After finding the PCI ISA device it is required to ask for the
resources to be read, before we use the base_addr for GPIO.

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
---
 src/geode_ddc.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/geode_ddc.c b/src/geode_ddc.c
index 2533488..086dbb1 100644
--- a/src/geode_ddc.c
+++ b/src/geode_ddc.c
@@ -72,6 +72,9 @@ geode_gpio_iobase(void)
     if (pci == NULL)
 	return 0;
 
+    if (pci_device_probe(pci) != 0)
+        return 0;
+
     /* The GPIO I/O address is in resource 1 */
     return (unsigned short)pci->regions[1].base_addr;
 #else
-- 
1.6.3.1.54.g99dd

